extensions/provisioninginput/machinepools.yml

```diff
@@ -0,0 +1,38 @@
+etcdControlPlaneWorker:
```

Comment:
Where these YAML files are used? Are these going to be used on r/r side?

Reply:
The ymls aren't going to be used on the rancher side but the functions that pull the data from the ymls will be used.

Reply:
Could you elaborate on this one for me, I don't think I understand this move yet, is there a code reference to these files that I am potentially missing?

Reply:
If you look in [pools.go](https://github.com/rancher/shepherd/blob/713005f72c344fa4afde5cb11d283383aecc261b/extensions/provisioninginput/pools.go) the 2 files are referenced by their respective functions. The main design goal here is to provide a way to store parts of our configs that are widely used and not sensitive without using global vars.

Reply:
Oh, I missed it, thanks for the reference. Could you get into more details about the design selection, what's the net positive implementation of having a YAML file, reading that file, and making it a Golang object, instead of using it as a Golang object?

Reply:
There are several positives:
1. Cleaner code. We use quite a few objects (which are basically yaml data) throughout the framework where we declare a default i.e machine pools, nodepools, group version resources... etc. When declaring these defaults we currently can declare them in place which is often not very reusable or as easily maintained (especially if its declared in place in multiple locations) or we can declare a global var (bad coding practice). Machine pools is a great example of this, it is widely used across the framework/rancher and we would have quite a large amount of duplicate code if we declared a Golang object in place. 
2. Simpler Configuration: This also opens the door to creating more defaults for our configs. If you look at machinepools there are a finite number of machine pool configurations that we test with, this is especially true in the case of our static tests. The same applies to other parts of provisioning input as well i.e machineConfigs, etcd... etc and while we cannot store all parts of the config as defaults (vpcID, credentials, amis...) we can store a lot of it and reduce the amount of configuration required to create clusters. 

Overall I think that putting the equivalent of yaml data in a .go file is not as reusable, maintainable, or modular as storing it in a yaml file and then loading it wherever you need it (which is a 1 liner). _yaml data in a yaml file, go code in a go file_

Reply:
2. I don't know if I agree with this at all. If we want to test the different configurations out we would be that in the test itself. 

So I don't have a huge problem with this, but I have two concerns. One it completely just copies over our load config func. 

The other why is only the machine Pools default yaml but not the other defaults not? There is no consistency. The purpose of these is to set the roles themselves, and honestly only for the tests we have. I feel like this shouldn't even be in shepherd but in the provisioning directory in rancher/rancher instead. These are defaults for our specific tests.

Reply:
After we get our actions vs extensions resolved. This will most likely move anyways

Reply:
I agree that test specific configurations belong in the test they are designed for, but when a configuration is used in a variety of tests I generally lean towards putting it in a place were it can be reused instead of redeclared in multiple tests. 

The load config function, while similar in the sense that it loads data into an object, is not the same as the LoadConfig function which points specifically at the cattle_test_config and uses the github.com/cresty/defaults package.

I don't know if I follow what you mean by "no consistency". Within the defaults directory it is consistent, anything that is widely used across rancher/shepherd that can be stored as a const is stored in the defaults dir and when we require const objects we load them from a yaml file. The only exception to this rule is the groupversionresources and that only occurs because it is an upstream k8s object without struct tags. 

For the rest of what you have mentioned I would say that **most** of the defaults dir will end up in rancher after actions vs extensions gets resolved.

Reply:
Agreed this will get resolved anyways, and I'm honestly fine with this.

Reply:
I do think the LoadDefault function should not be here then, and more in the pkg. What do you think?

Reply:
The consistency thing I was referring to, is why is this in a yaml where as the GroupVersions in functions. That is what I am talking about. I want to reiterate, my overall concern is that some thing is added to the yaml and we're not testing what we say we're testing. Does that make sense?

Reply:
Going to reiterate: If I could put the GroupVersions in a yaml I would however, I cannot because they are upstream k8s objects that do not have struct tags and as such I cannot simply unmarshal or change them.

Reply:
I am not sure I understand your final comment. Can you give me an example of something that of something that gets added to the yaml that would be a problem?